internal/util/sessionutil/session_util.go

```diff
@@ -487,3 +505,96 @@ func (s *Session) Registered() bool {
 	}
 	return b
 }
+
+func (s *Session) SetEnableActiveStandBy(enable bool) {
+	s.enableActiveStandBy = enable
+}
+
+func (s *Session) updateStandby(b bool) {
+	s.isStandby.Store(b)
+}
+
+func (s *Session) ProcessActiveStandBy(activateFunc func()) error {
+	if !s.enableActiveStandBy {
+		return nil
+	}
+	key := s.ServerName
+	primaryLockKey := path.Join(s.metaRoot, DefaultServiceRoot+"primaryLock/", key)
+	completeKey := path.Join(s.metaRoot, DefaultServiceRoot, s.ServerName)
+
+	// lock for the key to ensure only one active
+	lockFn := func() error {
+		// this ttl must no less than the TTL for register lease
+		session, err := concurrency.NewSession(s.etcdCli, concurrency.WithTTL(DefaultTTL))
+		if err != nil {
+			log.Debug("Fail to create session", zap.Error(err))
```

Comment:
Here and in other places: comments should start with lowercase letters. Also this should be an log.Error()